### ISSUE TITLE
warning: circular argument reference - score

### ISSUE THREAD

**Issue by ericbrooke** [assaf/vanity#259]
Occurs when I try to run rspec
```
/Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/experiment/ab_test.rb:305: warning: circular argument reference - score
Coverage report generated for RSpec to /Users/ericbrooke/dev/qcentral/coverage. 0.0 / 0.0 LOC (100.0%) covered.
/Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/adapters/abstract_adapter.rb:14:in `rescue in establish_connection': Could not find  in your load path (RuntimeError)
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/adapters/abstract_adapter.rb:11:in `establish_connection'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/playground.rb:376:in `establish_connection'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/playground.rb:357:in `establish_connection'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/playground.rb:460:in `autoconnect'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/playground.rb:51:in `initialize'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/playground.rb:478:in `new'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/playground.rb:478:in `playground'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/frameworks/rails.rb:4:in `load!'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/vanity-1.9.1/lib/vanity/frameworks.rb:7:in `block in <class:Plugin>'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:30:in `instance_exec'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:30:in `run'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:55:in `block in run_initializers'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/tsort.rb:226:in `block in tsort_each'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/tsort.rb:429:in `each_strongly_connected_component_from'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/tsort.rb:347:in `block in each_strongly_connected_component'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/tsort.rb:345:in `each'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/tsort.rb:345:in `call'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/tsort.rb:345:in `each_strongly_connected_component'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/tsort.rb:224:in `tsort_each'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/tsort.rb:203:in `tsort_each'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:54:in `run_initializers'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/application.rb:352:in `initialize!'
	from /Users/ericbrooke/dev/qcentral/config/environment.rb:5:in `<top (required)>'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:121:in `require'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:121:in `require'
	from /Users/ericbrooke/dev/qcentral/spec/rails_helper.rb:4:in `<top (required)>'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:121:in `require'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:121:in `require'
	from /Users/ericbrooke/dev/qcentral/spec/controllers/albums_controller_spec.rb:1:in `<top (required)>'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1226:in `load'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1226:in `block in load_spec_files'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1224:in `each'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1224:in `load_spec_files'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:97:in `setup'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:85:in `run'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:70:in `run'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:38:in `invoke'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/exe/rspec:4:in `<top (required)>'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/bin/rspec:23:in `load'
	from /Users/ericbrooke/.rbenv/versions/2.2.2/bin/rspec:23:in `<main>'

**Comment by phillbaker** [comment]
@ericbrooke can you give a little more context? Are you running rspec in a rails project that includes vanity? Can you share your vanity config please?

**Comment by ericbrooke** [comment]
@phillbaker thanks for your attention :-)

Yes rails project;

```
development:
  adapter: redis
  connection: redis://localhost:6379/0
test:
  collecting: false
production:
  adapter: redis
  connection: redis://<%= ENV["REDIS_USER"] %>:<%= ENV["REDIS_PASSWORD"] %>@<%= ENV["REDIS_HOST"] %>:<%= ENV["REDIS_PORT"] %>/0
```

**Comment by phillbaker** [comment]
This was fixed in https://github.com/assaf/vanity/commit/1b5bf18636caa5eae279e2aef2e24c923e63b141, which is released in the 2.0.0.beta branch, if you update to the latest release, you'll have that fix.

**Comment by ericbrooke** [comment]
Thanks @phillbaker the upgrade got rid of the scope issue into another #261
